ci/ray_ci/ray_docker_container.py

```diff
@@ -49,6 +49,8 @@ def run(self) -> None:
                 "python .buildkite/copy_files.py --destination docker_login",
             ]
             for alias in self._get_image_names():
+                if os.environ["RAYCI_SCHEDULE"] != "nightly" and "nightly" in alias:
+                    continue
```

Comment:
I think we should just edit `_should_upload()`? return `true` when `RAYCI_SCHEDULE == nightly`? it should also check if branch name is release brach or not.

Reply:
I think this is the logic we want to use:

- if not on postmerge, return false // it won't have privilege to upload even when it tries
- if upload flag is not on, return false
- if it is release branch, return true // we still need to do this until we have a better workflow on releasing, but hopefull with fewer cherrypicks in the future, this won't matter that much anyways.
- if it is master branch and `RAYCI_SCHEDULE == nightly`, return true // this PR's use case
- else, return false // breaking old behavior, which uploads on every master commit.

Reply:
Yes I think this logic makes sense. This should also not upload `postmerge` daytime run builds since we only want to upload once a day with a `night.date.sha` tag

Reply:
I've updated the code with new logic for `_should_upload` and also added unit tests for `should_upload`